### PR TITLE
Update docs about Automatic Prerendering

### LINKED ISSUE #7791: [Next 9.0.0] req/res are undefined in getInitialProps in _document.js
# Bug report

## Describe the bug

After having upgraded to v9 this morning we noticed that CSP was failing. Upon further investigation, it turned out that `ctx.req` and `ctx.res` were returning `undefined` in `_document.js` (or .ts) when calling `getInitialProps`.

This has been reproduced by combining the basic custom-document example as well as the custom-server example.

## To Reproduce

https://codesandbox.io/s/next9-getinitialprops-bug-f1lj7

## Expected behavior

`req` and `res` should have content.

## Screenshots

Server logs:
image

## System information

- OS: MacOS / Linux (codesandbox + node:10-alpine tested)
- server-side issue
- Version of Next.js: 9.0.0

### PR DESCRIPTION
**Pull request by Timer**
Closes https://github.com/zeit/next.js/issues/7791

### CODE REVIEW

File: packages/next/README.md [hunk 1]
```diff
@@ -1412,11 +1412,22 @@ export default MyApp
...
+
+A custom `<Document>` can also include `getInitialProps` for expressing asynchronous server-rendering data requirements.
+
+> **Note**: `<Document>`'s `getInitialProps` function is not called during client-side transitions,
+> nor when a page is [automatically prerendered](#automatic-prerendering).
+
+> **Note**: Make sure to check if `ctx.req` / `ctx.res` are defined in `getInitialProps`.
+> These variables will be `undefined` when a page is being statically exported for `next export` or [automatic prerendering (static optimization)](#automatic-prerendering).
+
+To use a custom `<Document>`, must create a file at `./pages/_document.js` and extend the `Document` class:
```
Comment (ijjk) [inline packages/next/README.md]:
```suggestion
To use a custom `<Document>`, you create a file at `./pages/_document.js` and extend the `Document` class:
```

File: packages/next/README.md [hunk 2]
```diff
@@ -1948,13 +1959,34 @@ module.exports = {
 
 ## Automatic Prerendering
 
-Next.js will automatically detect if a page is `static` at build time with `next build`, this is currently based on whether or not `getInitialProps` is set in the page, if set Next.js will assume that your page is not `static` and requires a server or lambda bundle.
+Next.js will automatically determine a page is static (and can be prerendered) if it has no has blocking data requirements.
```
Comment (ijjk) [inline packages/next/README.md]:
```suggestion
Next.js will automatically determine if a page is static (can be prerendered) and it has no blocking data requirements.
```

File: packages/next/README.md [hunk 3]
```diff
@@ -1948,13 +1959,34 @@ module.exports = {
...
+This determination is made by the absence of `getInitialProps` in both the page itself and a [custom `<App>`](#custom-app) (if present).
+
+If `getInitialProps` is present, Next.js will not prerender the page.
+Instead, Next.js will use its default behavior and render the page on-demand, per-request (meaning Server-Side Rendering).
+
+If `getInitialProps` is abset, Next.js will **statically optimize** your page automatically by prerendering it to static HTML.
```
Comment (ijjk) [inline packages/next/README.md]:
```suggestion
If `getInitialProps` is absent, Next.js will **statically optimize** your page automatically by prerendering it to static HTML.
```

File: packages/next/README.md [hunk 4]
```diff
@@ -1948,13 +1959,34 @@ module.exports = {
...
+
+This feature allows Next.js to emit hybrid applications that contain **both server-rendered and statically generated pages**.
+This ensures Next.js always emits applications that are **fast by default**.
+
+> **Note**: Statically generated pages are still reactive: Next.js will hydrate your application client-side to give it full interactivity.
+
+This feature provides many benefits.
+For example, optimized pages require no server-side computation and can be instantly streamed to the end-user from CDN locations.
+
+The result is an _ultra fast_ loading experience for your users.
```
Comment (ijjk) [inline packages/next/README.md]:
```suggestion
This results is an _ultra fast_ loading experience for your users.
```

File: packages/next/README.md (hunk 2 above)
Comment (Timer) [reply packages/next/README.md]: How about

> Next.js automatically determines that a page is static (can be prerendered) if it has no has blocking data requirements.

?

Comment (ijjk) [reply packages/next/README.md]: sounds good, updated the suggestion

**Review by ijjk — APPROVED** [review]

**Pull request merged**